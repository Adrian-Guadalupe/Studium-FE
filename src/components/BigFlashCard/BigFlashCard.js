import React, {useState} from 'react';
import styled from 'styled-components';
import {Card} from 'antd';
import PropTypes from 'prop-types';
import {darken} from 'polished';
import FlashCards from '../../svgComponents/FlashCards.js';
import {APP_VIEW_MOBILE} from '../../utilities/constants.js';

/**
 * Big Flash Card
 *
 * @component
 * @example
 * const flashCard = {question: "Question?", answer: "Answer"}
 * return (
 *  <BigFlashCard flashCard={flashCard} />
 *  )
 */
const BigFlashCard = ({flashCard, appView}) => {
  const [position, setPosition] = useState('front');
  
  const flipCard = () => {
    
    const newPos = position === 'front' ? 'back' : 'front';
    setPosition(newPos);
  };
  
  return (
    <StyledCardContainer data-testid="StyledCardContainer"
                         style={{position: 'relative'}} onClick={flipCard}>
      <StyledCard
        position={position}
        
        style={appView === APP_VIEW_MOBILE ?
          {width: '285px', height: '421.56px'} :
          {width: '352PX', height: '460px'}}
      >
        <CardText>
          {position === 'front' ? flashCard.question : flashCard.answer}
        </CardText>
      </StyledCard>
      <FlashCards appView={appView}/>
    </StyledCardContainer>
  
  );
};

BigFlashCard.prototypes = {
  flashCard: PropTypes.objectOf({
    question: PropTypes.string.isRequired, answer: PropTypes.string.isRequired,
  }).isRequired,
};

const StyledCardContainer = styled.div`
  height: 390px;
  width: 283px;
  position: relative;
`;

const StyledCard = styled(Card)`
  && {
    margin: 0px auto 0 auto;
    z-index: 2;
    background: ${props => props.position === 'front' ? '#FFFBF4' :
  darken(.1, '#4ea699')};
    color: ${props => (props.position === 'front' ? '#1b1414c9' : 'white')};

    border-radius: 11px;
    > .ant-card-body {
      min-height: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }
  transition: all 1s;
`;
const CardText = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  font-size: 20px;
  font-weight: 900;
  transition: all 1s;
`;
export default BigFlashCard;
