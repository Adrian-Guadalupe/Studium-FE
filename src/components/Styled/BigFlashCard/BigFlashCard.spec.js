import React from 'react';
import {
  customRender,
  getNodesByType,
  getByTestId,
  fireEvent,
  store,
  getByRole,
  logOutMessageOrDebug,
  getByText,
  toHaveTextContent,wait
} from '../../../utilities/test-utils';
import BigFlashCard from './BigFlashCard';
import {renderer, act, create} from 'react-test-renderer';
import { render } from '@testing-library/react';


/**
 * Test the big flash card
 */

describe('Big Flash Card', () => {
    test('position front', () => {
    const { container, getByText } = render(<BigFlashCard flashCard = {{question: "question", answer: "answer"}}/>)
    expect(getByText('question')).toBeInTheDocument()
  })
    test('position back', () => {
    const { container, getByTestId, queryByText, debug} = render(<BigFlashCard flashCard = {{question: "question", answer: "answer"}}/>)
    const card = getByTestId('StyledCardContainer')
    fireEvent.click(card)
    expect(queryByText('answer')).toBeInTheDocument()
    //debug(queryByText('answer'))
  })
});