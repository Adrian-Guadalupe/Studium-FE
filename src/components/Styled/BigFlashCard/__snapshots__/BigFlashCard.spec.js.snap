// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigFlashCard renders 1`] = `
<div
  className="sc-AxjAm fqUJNL"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="ant-card sc-AxirZ eRsMyF ant-card-bordered"
    onClick={[Function]}
    position="front"
    style={
      Object {
        "height": "421.56px",
        "width": "285px",
      }
    }
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="sc-AxiKw cGOXds"
      >
        question
      </div>
    </div>
  </div>
  <img
    src="Card2.jpg"
    style={
      Object {
        "height": "417px",
        "left": "5px",
        "position": "absolute",
        "top": "14px",
        "width": "276px",
        "zIndex": -1,
      }
    }
  />
  <img
    src="Card3.jpg"
    style={
      Object {
        "height": "417px",
        "left": "7px",
        "position": "absolute",
        "top": "22px",
        "width": "260px",
        "zIndex": -3,
      }
    }
  />
</div>
`;
